File: apps/frontend/lib/features/auth/domain/value_objects/client_kind.dart
/// Client platforms supported by the auth API.
enum ClientKind(
  /// The request value accepted by the backend.
  final String wireName,
) {
  /// A browser using an HTTP-only refresh cookie.
  web('web'),

  /// An iOS app receiving its refresh token in the response body.
  ios('ios'),

  /// An Android app receiving its refresh token in the response body.
  android('android'),
}

File: apps/backend/src/job_status_found/features/auth/domain/value_objects/client_kind.py
"""Kinds of clients that fix how refresh tokens are delivered."""

from enum import StrEnum


class ClientKind(StrEnum):
    """A client platform and its refresh-token delivery channel."""

    WEB = "web"
    """A browser, which receives the refresh token only in an HTTP-only cookie."""

    IOS = "ios"
    """An iOS app, which receives the refresh token in the response body."""

    ANDROID = "android"
    """An Android app, which receives the refresh token in the response body."""
